core/src/main/java/org/apache/calcite/sql2rel/RelFieldTrimmer.java

```diff
@@ -697,14 +697,20 @@ public TrimResult trimFields(
 
   /**
    * Variant of {@link #trimFields(RelNode, ImmutableBitSet, Set)} for
-   * {@link org.apache.calcite.rel.core.SetOp} (including UNION and UNION ALL).
```

Comment:
I think a doc here to explain why not trim other SetOp types might be helpful for others.